Sources/BraveWallet/Crypto/Transactions/TransactionDetailsView.swift

```diff
@@ -14,122 +14,569 @@ struct TransactionDetailsView: View {
+
+private struct TransactionStatusBadgeView: View {
   
-  private func detailRow(title: String, value: String) -> some View {
-    detailRow(title: title) {
-      Text(value)
-        .multilineTextAlignment(.trailing)
-    }
-  }
+  let status: BraveWallet.TransactionStatus
   
-  private func detailRow<ValueView: View>(title: String, @ViewBuilder valueView: () -> ValueView) -> some View {
+  var body: some View {
     HStack {
-      Text(title)
-      Spacer()
-      valueView()
+      if status.shouldShowLoadingStatus {
+        ProgressView()
+          .progressViewStyle(.braveCircular(size: .mini))
+      } else if status.shouldShowSuccessStatus {
+        Image(braveSystemName: "leo.check.circle-outline")
+          .foregroundColor(Color(braveSystemName: .systemfeedbackSuccessIcon))
+      } else if status.shouldShowErrorStatus {
+        Image(braveSystemName: "leo.warning.circle-outline")
+          .foregroundColor(Color(braveSystemName: .systemfeedbackSuccessIcon))
+      }
+      Text(status.localizedDescription)
+        .foregroundColor(status.badgeTextColor)
     }
     .font(.caption)
-    .foregroundColor(Color(.braveLabel))
-    .padding(.horizontal)
-    .padding(.vertical, 12)
-    .listRowBackground(Color(.secondaryBraveGroupedBackground))
+    .padding(.horizontal, 6)
+    .padding(.vertical, 4)
+    .background(
+      RoundedRectangle(cornerRadius: 4)
+        .fill(status.badgeBackgroundColor)
+    )
+  }
+}
+
+private extension BraveWallet.TransactionStatus {
+  /// If we should show transaction status as loading
+  var shouldShowLoadingStatus: Bool {
+    switch self {
+    case .unapproved, .submitted:
+      return true
+    default:
+      return false
+    }
+  }
+  
+  /// If we should show transaction status successful icon
+  var shouldShowSuccessStatus: Bool {
+    switch self {
+    case .approved, .confirmed, .signed:
+      return true
+    default:
+      return false
+    }
+  }
+  
+  /// If we should show transaction status failure icon
+  var shouldShowErrorStatus: Bool {
+    switch self {
+    case .error, .dropped, .rejected:
+      return true
+    default:
+      return false
+    }
+  }
+  
+  /// Color of status text on status badge
+  var badgeTextColor: Color {
+    switch self {
+    case .confirmed, .approved:
+      return Color(braveSystemName: .systemfeedbackSuccessText)
+    case .rejected, .error, .dropped:
+      return Color(braveSystemName: .systemfeedbackErrorText)
+    case .unapproved:
+      return Color(braveSystemName: .textSecondary)
+    case .submitted, .signed:
+      return Color(braveSystemName: .systemfeedbackInfoText)
+    @unknown default:
+      return Color.clear
+    }
+  }
+  
+  /// Color of status badge
+  var badgeBackgroundColor: Color {
+    switch self {
+    case .confirmed, .approved:
+      return Color(braveSystemName: .systemfeedbackSuccessBackground)
+    case .rejected, .error, .dropped:
+      return Color(braveSystemName: .systemfeedbackErrorIcon)
```

Comment:
im not sure if this is the correct color for those three statuses. looks too red.
![Simulator Screenshot - iPhone 15 Pro - 2023-11-27 at 14 51 34](https://github.com/brave/brave-ios/assets/1187676/07227a11-7d0a-4e32-be2a-bdb17b294b1e)

Reply:
Good catch, had wrong colour for failed/rejected/dropped status. Fixed in [ff80cea](https://github.com/brave/brave-ios/pull/8474/commits/ff80cead166e23a009d8ce02410d01c89343c439).
![Simulator Screenshot - iPhone 15 Pro - 2023-11-28 at 09 53 32](https://github.com/brave/brave-ios/assets/5314553/f82762bd-c3dc-43dc-91ea-9a2a6a4b8d3b)